tests/components/common/pills/pills.spec.tsx

```diff
@@ -0,0 +1,68 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { Pills } from "src/components/common/pills";
+import { EElementType, TElement } from "src/context-providers";
+import { ELEMENT_BUTTON_LABELS } from "src/data";
+import { TestHelper } from "src/util/test-helper";
+
+describe("Pills", () => {
+    beforeEach(() => {
+        global.ResizeObserver = jest.fn().mockImplementation(() => ({
+            observe: jest.fn(),
+            unobserve: jest.fn(),
+            disconnect: jest.fn(),
+        }));
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+
+    it("should render the pill items and list position field if pills field is toggled from false to true", async () => {
```

Comment:
split into multiple tests to narrow the scope of each test
- test that pills are false by default and no pill items are rendered
- test for conditional rendering of pill items